ui/src/components/Settings/Notifications/CreateNotificationModal/index.tsx

```diff
@@ -0,0 +1,302 @@
+      (status) => {
+        props.onSave(status)
+      },
+    )
+  }
+
+  const handleInputChange = (fieldName: string, value: any) => {
+    setFormValues((prevValues: any) => ({
+      ...prevValues,
+      [fieldName]: value,
+    }))
+  }
+
+  if (!availableAgents || !availableTypes) {
+    return (
+      <span>
+        <LoadingSpinner />
+      </span>
+    )
+  } else {
+    return (
+      <Modal
+        loading={false}
+        backgroundClickable={false}
+        onCancel={() => props.onCancel()}
+        okDisabled={false}
+        okText="Save"
+        okButtonType={'primary'}
+        title={'New Notification'}
+        iconSvg={''}
+        onOk={handleSubmit}
+      >
+        <div>
+          <form className="space-y-4">
+            {/* Config Name */}
+            <div className="form-row">
+              <label htmlFor="name" className="text-label">
+                Name *
+              </label>
+              <div className="form-input">
+                <div className="form-input-field">
+                  <input
+                    type="text"
+                    name="name"
+                    defaultValue={props.selected?.name}
+                    onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
+                      (nameRef.current = event.target.value)
+                    }
+                  ></input>
+                </div>
+              </div>
+            </div>
+            {/* Enabled */}
+            <div className="form-row">
+              <label htmlFor="enabled" className="text-label">
+                Enabled
+              </label>
+              <div className="form-input">
+                <div className="form-input-field">
+                  <input
+                    type="checkbox"
+                    name="enabled"
+                    id="enabled"
+                    defaultChecked={props.selected?.enabled}
+                    onChange={(event: React.ChangeEvent<HTMLInputElement>) =>
+                      event.target.value === 'on'
+                        ? (enabledRef.current = true)
+                        : (enabledRef.current = false)
+                    }
+                  ></input>
+                </div>
+              </div>
+            </div>
+            {/* Select agent */}
+            <div className="form-row">
+              <label htmlFor="ssl" className="text-label">
+                Agent *
+              </label>
+              <div className="form-input">
+                <div className="form-input-field">
+                  <select
+                    id="agent"
+                    name="agent"
+                    onChange={(e) => {
+                      setFormValues({})
+                      setTargetAgent(availableAgents[Number(e.target.value)])
+                    }}
+                    className="rounded-l-only"
+                  >
+                    {availableAgents?.map((agent, index) => (
```

Comment:
Does the 'none' option need filtering out here, or just be made into an empty string?

Reply:
I was planning to remove the 'none' option entirely since it's not being used anywhere.

Adding a friendly name mapping sounds like a great idea.